examples/metadata-indexer/src/indexerQueue.ts

```diff
@@ -117,65 +135,82 @@ export class IndexerQueue {
     this.logProgress();
   }
 
-  private async indexMetadata(url: string, result: any) {
+  private async indexMetadata(url: string, result: any | null) {
+    if (!result) {
+      await this.db
+        .insertInto("urlMetadata")
+        .values({ updatedAt: new Date(), url })
+        .onConflict((oc) => {
+          return oc.columns(["url"]).doNothing();
+        })
+        .execute();
+      return;
+    }
+
     const nftCollectionId = result.nft?.collection.id;
     const nftMetadataId = result.nft?.tokenId
       ? `${result.nft?.collection.id}/${result.nft?.tokenId}`
       : null;
 
+    const urlMetadataValue = {
+      url,
+      customOpenGraph: result.customOpenGraph ?? null,
+      updatedAt: new Date(),
+      imageUrl: result.image?.url,
+      imageWidth: result.image?.width,
+      imageHeight: result.image?.height,
+      description: result.description,
+      alt: result.alt,
+      title: result.title,
+      publisher: result.publisher,
+      logoUrl: result.logo?.url,
+      mimeType: result.mimeType,
+      nftCollectionId: nftCollectionId,
+      nftMetadataId: nftMetadataId,
+    };
+
     await this.db
       .insertInto("urlMetadata")
-      .values({
-        url,
-        customOpenGraph: result.customOpenGraph ?? null,
-        updatedAt: new Date(),
-        imageUrl: result.image?.url,
-        imageWidth: result.image?.width,
-        imageHeight: result.image?.height,
-        description: result.description,
-        alt: result.alt,
-        title: result.title,
-        publisher: result.publisher,
-        logoUrl: result.logo?.url,
-        mimeType: result.mimeType,
-        nftCollectionId: nftCollectionId,
-        nftMetadataId: nftMetadataId,
+      .values(urlMetadataValue)
+      .onConflict((oc) => {
+        return oc.columns(["url"]).doUpdateSet(urlMetadataValue);
```

Comment:
does this do a JSON nested update or not? if it does, maybe it shouldn't, or else removing a button won't update. I suspect it doesn't